components/director/internal/domain/labeldef/resolver.go

```diff
@@ -64,3 +66,60 @@ func (r *Resolver) CreateLabelDefinition(ctx context.Context, in graphql.LabelDe
 
 	return &out, nil
 }
+
+func (r *Resolver) LabelDefinitions(ctx context.Context) ([]*graphql.LabelDefinition, error) {
+	tnt, err := tenant.LoadFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	tx, err := r.transactioner.Begin()
+	if err != nil {
+		return nil, errors.Wrap(err, "while starting transaction")
+	}
+	defer r.transactioner.RollbackUnlessCommited(tx)
+	ctx = persistence.SaveToContext(ctx, tx)
+
+	defs, err := r.srv.List(ctx, tnt)
+	if err != nil {
+		return nil, errors.Wrap(err, "while listing Label Definitions")
+	}
+
+	if err := tx.Commit(); err != nil {
+		return nil, errors.Wrap(err, "while committing transaction")
+	}
+
+	var out []*graphql.LabelDefinition
+	for _, def := range defs {
+		c := r.conv.ToGraphQL(def)
+		out = append(out, &c)
+	}
+	return out, nil
+}
+
+func (r *Resolver) LabelDefinition(ctx context.Context, key string) (*graphql.LabelDefinition, error) {
+	tnt, err := tenant.LoadFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := r.transactioner.Begin()
+	if err != nil {
+		return nil, errors.Wrap(err, "while starting transaction")
+	}
+	defer r.transactioner.RollbackUnlessCommited(tx)
+	ctx = persistence.SaveToContext(ctx, tx)
+	def, err := r.srv.Get(ctx, tnt, key)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "while getting Label Definition")
+	}
+	if err := tx.Commit(); err != nil {
+		return nil, errors.Wrap(err, "while committing transaction")
+	}
+
+	if def == nil {
```

Comment:
Shouldn' we return error if not found?

Reply:
I believe we shouldn't, as it is top level query.

Reply:
for now, we return an error to be consistent with other endpoints, but follow up task will be created